src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1105,11 +1105,13 @@ class DocBaseViewer extends BaseViewer {
         this.controls.add(__('exit_fullscreen'), this.toggleFullscreen, 'bp-exit-fullscreen-icon', ICON_FULLSCREEN_OUT);
 
         if (this.areNewAnnotationsEnabled() && this.hasAnnotationCreatePermission()) {
+            const canDownload = checkPermission(this.options.file, PERMISSION_DOWNLOAD);
```

Comment:
Should we use the `canDownload` util from lib/file.js?

Reply:
Seems like `canDownload` from `lib/file.js` checks a lot more than just the download permission related to actually being able to download the file. Might be too restrictive?

Reply:
Yeah, fair point. It also looks like we only check the permission for `textLayerMode`. Should be good to go here.